cmd/lotus-shed/balances.go

```diff
@@ -941,3 +949,335 @@ var fillBalancesCmd = &cli.Command{
+				types.FIL(pledge).String(),
+				types.FIL(oldPledge).String(),
+				types.FIL(types.BigSub(pledge, oldPledge)).String(),
+			)
+		}
+		fmt.Println(strings.Repeat("-", 119))
+
+		return nil
+	},
+}
+
+// from itests/migration_test.go
+
+// preFip0081StateMinerInitialPledgeForSector is the same calculation as StateMinerInitialPledgeForSector
+// but uses miner14's version of the calculation without the FIP-0081 changes.
+func preFip0081StateMinerInitialPledgeForSector(
+	ctx context.Context,
+	client api.FullNode,
+	sectorDuration abi.ChainEpoch,
+	sectorSize abi.SectorSize,
+	verifiedSize uint64,
+	ts *types.TipSet,
+) (types.BigInt, error) {
+	bs := blockstore.NewAPIBlockstore(client)
+	ctxStore := gststore.WrapBlockStore(ctx, bs)
+
+	circSupply, err := client.StateVMCirculatingSupplyInternal(ctx, ts.Key())
+	if err != nil {
+		return types.NewInt(0), err
+	}
+
+	powerActor, err := client.StateGetActor(ctx, power.Address, ts.Key())
+	if err != nil {
+		return types.NewInt(0), err
+	}
+
+	powerState, err := power.Load(ctxStore, powerActor)
+	if err != nil {
+		return types.NewInt(0), err
+	}
+
+	rewardActor, err := client.StateGetActor(ctx, reward.Address, ts.Key())
+	if err != nil {
+		return types.NewInt(0), err
+	}
+
+	rewardState, err := reward.Load(ctxStore, rewardActor)
+	if err != nil {
+		return types.NewInt(0), err
+	}
+
+	networkQAPower, err := powerState.TotalPowerSmoothed()
+	if err != nil {
+		return types.NewInt(0), err
+	}
+
+	verifiedWeight := big.Mul(big.NewIntUnsigned(verifiedSize), big.NewInt(int64(sectorDuration)))
+	sectorWeight := builtin.QAPowerForWeight(sectorSize, sectorDuration, verifiedWeight)
+
+	thisEpochBaselinePower, err := rewardState.(interface {
+		ThisEpochBaselinePower() (abi.StoragePower, error)
+	}).ThisEpochBaselinePower()
+	if err != nil {
+		return types.NewInt(0), err
+	}
+	thisEpochRewardSmoothed, err := rewardState.(interface {
+		ThisEpochRewardSmoothed() (builtin.FilterEstimate, error)
+	}).ThisEpochRewardSmoothed()
+	if err != nil {
+		return types.NewInt(0), err
+	}
+
+	rewardEstimate := smoothing14.FilterEstimate{
+		PositionEstimate: thisEpochRewardSmoothed.PositionEstimate,
+		VelocityEstimate: thisEpochRewardSmoothed.VelocityEstimate,
+	}
+	networkQAPowerEstimate := smoothing14.FilterEstimate{
+		PositionEstimate: networkQAPower.PositionEstimate,
+		VelocityEstimate: networkQAPower.VelocityEstimate,
+	}
+
+	initialPledge := miner14.InitialPledgeForPower(
+		sectorWeight,
+		thisEpochBaselinePower,
+		rewardEstimate,
+		networkQAPowerEstimate,
+		circSupply.FilCirculating,
+	)
+
+	var initialPledgeNum = types.NewInt(110)
+	var initialPledgeDen = types.NewInt(100)
+
+	return types.BigDiv(types.BigMul(initialPledge, initialPledgeNum), initialPledgeDen), nil
+}
+
+// postFip0081StateMinerInitialPledgeForSector should be the same calculation as StateMinerInitialPledgeForSector,
+// it's here for sanity checking.
+func postFip0081StateMinerInitialPledgeForSector(
```

Comment:
This function is not strictly necessary, I added it in as a sanity check to follow the same logic as `preFip0081StateMinerInitialPledgeForSector` but it should arrive at the post-FIP-0081 result.
I can remove this, and the sanity check it uses, if it feels crufty, because it doesn't contribute much.